src/MALA.jl

```diff
@@ -85,9 +82,7 @@ end
 Return the value and gradient of the log density of the parameters `params` for the `model`.
 """
 function logdensity_and_gradient(model::DensityModel, params)
-    res = GradientResult(params)
-    gradient!(res, model.logdensity, params)
-    return value(res), gradient(res)
+    error("no implementation exist for `DensityModel`; try importing ForwardDiff.jl")
```

Comment:
Maybe let's also throw an error below if it is called for `model::LogDensityModel` with `LogDensityProblems.capabilities(model.logdensity) === LogDensityProblems.LogDensityOrder{0}()`?

Reply:
Won't LogDensityProblems.jl do that?

Reply:
No: https://github.com/tpapp/LogDensityProblems.jl/blob/69a014986d5cacaa483e63b5b9b228886e82ebb8/src/LogDensityProblems.jl#L128 Maybe it should?

Reply:
Aaah I was mixing it up with `ADgradient` which complains if you haven't imported the desired package yet; added a check now:)

> Maybe it should?

Is that more informative that telling the user that we're missing a method definition `logdensity_and_gradient`? Not sure :shrug:

Reply:
No, me neither.